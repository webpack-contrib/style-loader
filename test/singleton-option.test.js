import compile from './helpers/compiler';
import runTestInJsdom from './helpers/runTestInJsdom';

describe('singleton option', () => {
  it('not specified', async () => {
    expect.assertions(3);

    const testId = './singleton.js';
    const stats = await compile(testId);

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('false', async () => {
    expect.assertions(3);

    const testId = './singleton.js';
    const stats = await compile(testId, {
      loader: {
        options: {
          singleton: false,
        },
      },
    });

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });

  it('true', async () => {
    expect.assertions(3);

    const testId = './singleton.js';
    const stats = await compile(testId, {
      loader: {
        options: {
          singleton: true,
        },
      },
    });

    runTestInJsdom(stats, (dom) => {
      expect(dom.serialize()).toMatchSnapshot('DOM');
    });

    expect(stats.compilation.warnings).toMatchSnapshot('warnings');
    expect(stats.compilation.errors).toMatchSnapshot('errors');
  });
});
