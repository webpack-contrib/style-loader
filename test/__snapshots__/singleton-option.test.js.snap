// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`singleton option false: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">.class-one {
  color: red;
}
</style><style type=\\"text/css\\">.class-two {
  color: red;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`singleton option false: errors 1`] = `Array []`;

exports[`singleton option false: warnings 1`] = `Array []`;

exports[`singleton option not specified: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">.class-one {
  color: red;
}
</style><style type=\\"text/css\\">.class-two {
  color: red;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`singleton option not specified: errors 1`] = `Array []`;

exports[`singleton option not specified: warnings 1`] = `Array []`;

exports[`singleton option true: DOM 1`] = `
"<!DOCTYPE html><html><head>
  <title>style-loader test</title>
  <style id=\\"existing-style\\">.existing { color: yellow }</style>
<style type=\\"text/css\\">.class-one {
  color: red;
}
.class-two {
  color: red;
}
</style></head>
<body>
  <h1>Body</h1>
  <div class=\\"target\\"></div>
  <iframe class=\\"iframeTarget\\"></iframe>


</body></html>"
`;

exports[`singleton option true: errors 1`] = `Array []`;

exports[`singleton option true: warnings 1`] = `Array []`;
